cloudinit/net/netplan.py

```diff
@@ -149,8 +164,14 @@ def _listify(obj, token=" "):
                     "via": route.get("gateway"),
                     "to": to_net,
                 }
-                if "metric" in route:
-                    new_route.update({"metric": route.get("metric", 100)})
+                # Priority for metric: 1. route's metric, 2. subnet's metric, 3. default 100
+                route_metric = route.get("metric")
+                if route_metric is not None:
+                    new_route["metric"] = route_metric
+                elif sn_metric is not None:
+                    new_route["metric"] = sn_metric
+                else:
+                    new_route["metric"] = 100
```

Comment:
My mistake for suggesting this fallback to a default of 100 when no metric key is presented. This could cause some significant issues with more complex networks if cloud-init decides to set metric of 100 on routes when no `metric` key is provided. I think we should drop that default and only add metric of 100 if specifically provided. Looking over the previous code, it never would have triggered the default of 100 because `route.get()` was only called when "metric" was present in network-config. I've pushed a supplemental commit which drops this unintentional default when no `metric` key is present and fixes the tests.